apps/ejabberd/src/ejabberd_auth_jwt.erl

```diff
@@ -0,0 +1,215 @@
+                    false;
+                error ->
+                    ?WARNING_MSG("Missing key ~p in JWT ~p~n", [UserKey, TokenData]),
+                    false
+            end;
+        {error, Reason} ->
+            ?WARNING_MSG("Cannot verify JWT for user ~s: ~p~n", [LUser, Reason]),
+            false
+    end.
+
+
+-spec check_password(LUser :: ejabberd:luser(),
+                     LServer :: ejabberd:lserver(),
+                     Password :: binary(),
+                     Digest :: binary(),
+                     DigestGen :: fun()) -> boolean().
+check_password(LUser, LServer, Password, _Digest, _DigestGen) ->
+    check_password(LUser, LServer, Password).
+
+
+-spec set_password(LUser :: ejabberd:luser(),
+                   LServer :: ejabberd:lserver(),
+                   Password :: binary()) -> ok | {error, not_allowed | invalid_jid}.
+set_password(_LUser, _LServer, _Password) ->
+    {error, not_allowed}.
+
+
+-spec try_register(LUser :: ejabberd:luser(),
+                   LServer :: ejabberd:lserver(),
+                   Password :: binary()
+                   ) -> ok | {error, exists | not_allowed}.
+try_register(_LUser, _LServer, _Password) ->
+    {error, not_allowed}.
+
+
+-spec dirty_get_registered_users() -> [ejabberd:simple_bare_jid()].
+dirty_get_registered_users() ->
+    [].
+
+
+-spec get_vh_registered_users(LServer :: ejabberd:lserver()
+                             ) -> [ejabberd:simple_bare_jid()].
+get_vh_registered_users(_LServer) ->
+    [].
+
+
+-type query_keyword() :: from | to | limit | offset | prefix.
+-type query_value() :: integer() | binary().
+-spec get_vh_registered_users(LServer :: ejabberd:lserver(),
+                              Query :: [{query_keyword(), query_value()}]
+                              ) -> [ejabberd:simple_bare_jid()].
+get_vh_registered_users(LServer, _) ->
+    get_vh_registered_users(LServer).
+
+
+-spec get_vh_registered_users_number(LServer :: ejabberd:server()
+                                    ) -> non_neg_integer().
+get_vh_registered_users_number(_LServer) ->
+    0.
+
+
+-spec get_vh_registered_users_number(LServer :: ejabberd:lserver(),
+                                     Query :: [{prefix, binary()}]
+                                     ) -> integer().
+get_vh_registered_users_number(LServer, _) ->
+    get_vh_registered_users_number(LServer).
+
+
+-spec get_password(LUser :: ejabberd:luser(),
+                   LServer :: ejabberd:lserver()) -> binary() | false.
+get_password(_LUser, _LServer) ->
+    false.
+
+
+-spec get_password_s(LUser :: ejabberd:luser(),
+                     LServer :: ejabberd:lserver()) -> binary().
+get_password_s(_LUser, _LServer) ->
+    <<"">>.
+
+-spec does_user_exist(LUser :: ejabberd:luser(),
+                     LServer :: ejabberd:lserver()
+                     ) -> boolean() | {error, atom()}.
+does_user_exist(_LUser, _LServer) ->
+    true.
```

Comment:
Are we sure there are no implicit negative consequences of always returning `true` here?